src/parser/postgresparser.cpp

```diff
@@ -1408,20 +1415,19 @@ parser::SQLStatementList* PostgresParser::ParseSQLString(const char* text) {
   auto result = pg_query_parse(text);
   if (result.error) {
     // Parse Error
-    LOG_ERROR("%s at %d\n", result.error->message, result.error->cursorpos);
-    auto new_stmt = new parser::SQLStatementList();
-    new_stmt->is_valid = false;
+    std::string exception_msg = StringUtil::Format(
+        "%s at %d", result.error->message, result.error->cursorpos);
     pg_query_parse_finish(ctx);
     pg_query_free_parse_result(result);
-    return new_stmt;
+    throw ParserException(exception_msg);
```

Comment:
Is this the only major change? The rest is just reformatting?

Reply:
Yes, I played with that and found that the problem was that we returned a list whose valid flag was false without throwing an exception, so if we just change that we'll be good. I guess why this problem was never discovered was that we never threw queries with wrong syntax on the parser before.